tests/subcommands.rs

```diff
@@ -211,3 +205,93 @@ fn flatten_enum() {
         }
     );
 }
+
+#[test]
+fn external_subcommand_string() {
+    #[derive(StructOpt, Debug, PartialEq)]
+    struct Opt {
+        #[structopt(flatten)]
+        sub_cmd: SubCmd,
+    }
+    #[derive(StructOpt, Debug, PartialEq)]
+    enum SubCmd {
+        Foo {
+            #[structopt(long)]
+            flag: bool,
+            #[structopt(long)]
+            arg: u32,
+        },
+
+        #[structopt(external_subcommand)]
+        Other(Vec<String>),
+    }
+
+    assert_eq!(
+        Opt::from_iter(&["test", "foo", "--flag", "--arg=0"]),
+        Opt {
+            sub_cmd: SubCmd::Foo { flag: true, arg: 0 }
+        }
+    );
+
+    assert_eq!(
+        Opt::from_iter(&["test", "bar", "--flag", "--arg=0"]),
+        Opt {
+            sub_cmd: SubCmd::Other(vec!["bar".into(), "--flag".into(), "--arg=0".into()])
+        }
+    );
+    assert_eq!(
+        Opt::from_iter(&["test"]),
+        Opt {
+            sub_cmd: SubCmd::Other(vec!["".into()])
+        }
```

Comment:
What's the behavior should be in this case? We could pass `SubcommandRequiredOrHelp` along with `AllowExternalSubcommands`

Reply:
Agree, without option on the subcommand, the subcommand must be provided, and that's not the case here.